src/frontend/src/session.rs

```diff
@@ -937,95 +938,103 @@ pub struct SessionManagerImpl {
+                            let mut salt = [0; 4];
+                            let mut rng = rand::thread_rng();
+                            rng.fill_bytes(&mut salt);
+                            UserAuthenticator::Md5WithSalt {
+                                encrypted_password: md5_hash_with_salt(
+                                    &auth_info.encrypted_value,
+                                    &salt,
+                                ),
+                                salt,
+                            }
+                        } else if auth_info.encryption_type == EncryptionType::Oauth as i32 {
+                            UserAuthenticator::OAuth(auth_info.metadata.clone())
+                        } else {
+                            return Err(Box::new(Error::new(
+                                ErrorKind::Unsupported,
+                                format!("Unsupported auth type: {}", auth_info.encryption_type),
+                            )));
                         }
-                    } else if auth_info.encryption_type == EncryptionType::Oauth as i32 {
-                        UserAuthenticator::OAuth(auth_info.metadata.clone())
-                    } else {
-                        return Err(Box::new(Error::new(
-                            ErrorKind::Unsupported,
-                            format!("Unsupported auth type: {}", auth_info.encryption_type),
-                        )));
                     }
-                }
-            };
+                };
 
-            // Assign a session id and insert into sessions map (for cancel request).
-            let secret_key = self.number.fetch_add(1, Ordering::Relaxed);
-            // Use a trivial strategy: process_id and secret_key are equal.
-            let id = (secret_key, secret_key);
-            let session_impl: Arc<SessionImpl> = SessionImpl::new(
-                self.env.clone(),
-                Arc::new(AuthContext::new(
+                let auth_ctx = Arc::new(AuthContext::new(
                     database.to_string(),
                     user_name.to_string(),
                     user.id,
-                )),
-                user_authenticator,
-                id,
-                peer_addr,
-            )
-            .into();
-            self.insert_session(session_impl.clone());
-
-            Ok(session_impl)
-        } else {
-            Err(Box::new(Error::new(
-                ErrorKind::InvalidInput,
-                format!("Role {} does not exist", user_name),
-            )))
-        }
+                ));
+                (auth_ctx, user_authenticator)
+            } else {
+                return Err(Box::new(Error::new(
+                    ErrorKind::InvalidInput,
+                    format!("Role {} does not exist", user_name),
+                )));
+            }
+        };
+        let session_config = self.env.meta_client().get_session_params().await?;
```

Comment:
This looks bad. Should we adopt the notification/subscription mechanism just like the catalogs?

Reply:
This is how session variable works in pg

Reply:
Maybe it is enough to only load the config when the frontend starts for now?
https://www.postgresql.org/docs/current/sql-altersystem.html

> Values set with ALTER SYSTEM will be effective after the next server configuration reload, or after the next server restart in the case of parameters that can only be changed at server start. A server configuration reload can be commanded by calling the SQL function pg_reload_conf(), running pg_ctl reload, or sending a SIGHUP signal to the main server process.

Reply:
The reload behavior of different config depends on its `context` value, according to https://www.postgresql.org/docs/current/view-pg-settings.html. We may verify the behavior for calling `ALTER SYSTEM` on these session configs in Postgres and provide a precise definition then.

Reply:
We have no plan to support config file for session parameters cc @fuyufjh

Reply:
> We have no plan to support config file for session parameters

Yes. Different from PG, RW is distributed, which naturally makes config file different from the system params.